### PR TITLE
Adding `Environment` module for running with mock environment variables.

### PR DESCRIPTION
**Pull request by tmbull**
I created this `Environment` module for use in a project that is using `mockery`, and I thought you guys might find it useful.

### CODE REVIEW

File: mockery.cabal [hunk 1]
```diff
@@ -1,9 +1,9 @@
--- This file has been generated from package.yaml by hpack version 0.5.4.
+-- This file has been generated from package.yaml by hpack version 0.9.0.
 --
 -- see: https://github.com/sol/hpack
 
 name:           mockery
-version:        0.3.2
+version:        0.3.1
```
Comment (tmbull) [inline mockery.cabal]: I thought about omitting this line, but I think it is useful to point out that the version in the cabal file did not match the version in the package.yaml, which suggests that the cabal file was modified directly.

**Comment by tmbull** [comment]
Hmm it looks like you guys are still using GHC 7.6.3, which does not have support for the `setEnv` function. If you do not intend to upgrade GHC, then you will probably just want to close this PR as I don't see a way around it for older GHCs. Otherwise, please let me know what I can do to help.

**Comment by tmbull** [comment]
At @soenkehahn's suggestion, I used base-compat to solve the issue with GHC 7.6.3. Please let me know if there is anything else you need from me.

**Comment by sol** [comment]
LGTM.

@tmbull Can you rebase on current `master` an squash your commits into a single commit, please?  I'll then apply and release to Hackage.

**Comment by tmbull** [comment]
@sol I've cleaned up the PR as you suggested. Thanks!

**Pull request merged**